packages/next-bundle-analyzer/index.js

```diff
@@ -1,19 +1,26 @@
 module.exports =
-  ({ enabled = true, openAnalyzer = true } = {}) =>
+  (bundleAnalyzerConfig = {}) =>
   (nextConfig = {}) => {
     return Object.assign({}, nextConfig, {
       webpack(config, options) {
-        if (enabled) {
+        const analyzerConfig =
+          typeof bundleAnalyzerConfig === 'function'
```

Comment:
Accepting a function allows consumer to set any config they would like and it can even depend on client/server. You could even enable/disable based on config/options, so you could generate just a client or server bundle